### PR TITLE
Add test coverage for github-event

### PR DESCRIPTION
**Pull request by ivanayov**
This adds tests for github-event handler. Covers empty/non-suported event, validate customers and push event without secrets

Signed-off-by: Ivana Yovcheva <iyovcheva@vmware.com>

Needs SDK release from #123 for secrets in order to proceed with push and installation events.

## How Has This Been Tested?

Unit tested

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required) N/A
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [x] added unit tests

### CODE REVIEW

File: github-event/handler.go [hunk 1]
```diff
@@ -18,9 +18,19 @@ import (
 // Source name for this function when auditing
 const Source = "github-event"
 
+var (
+	audit  sdk.Audit
+	client = &http.Client{}
```
Comment (alexellis) [inline github-event/handler.go]: Please can you switch to using `http.DefaultClient` instead?

File: github-event/handler.go [hunk 2]
```diff
@@ -18,9 +18,19 @@ import (
...
+)
+
 // Handle receives events from the GitHub app and checks the origin via
 // HMAC. Valid events are push or installation events.
 func Handle(req []byte) string {
+
+	if audit == nil {
+		audit = sdk.AuditLogger{}
```
Comment (alexellis) [inline github-event/handler.go]: 👍

File: github-event/handler.go [hunk 3]
```diff
@@ -309,8 +315,8 @@ func getCustomers(customerURL string) ([]string, error) {
 
 		customers = strings.Split(string(pageBody), "\n")
 
-		for i, c := range customers {
-			customers[i] = strings.ToLower(strings.TrimSuffix(c, "\r"))
+		for i, client := range customers {
```
Comment (alexellis) [inline github-event/handler.go]: Can this be `customer` please?

File: github-event/handler_test.go [hunk 4]
```diff
@@ -63,3 +65,74 @@ func Test_Handle_ValidateCustomersInvalid(t *testing.T) {
...
+	var events = []struct {
+		title             string
+		header            string
+		action            string
+		validateCustomers string
+		validateHmac      string
+		want              string
+	}{
+		{
+			title:             "Empty event",
```
Comment (alexellis) [inline github-event/handler_test.go]: Please update from title to scenario

**Comment by ivanayov** [comment]
@alexellis addressed your comments

**Comment by alexellis** [comment]
Thank you

**Comment by alexellis** [comment]
@ivanayov I'm publishing a change for 0.7.2 of this function. Please raise a PR against orc-bootstrap to push your change into the too.

**Comment by ivanayov** [comment]
@alexellis ok, I'll open one there

**Pull request merged**